addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js

```diff
@@ -448,6 +457,204 @@ QUnit.test('chatter should become enabled when creation done', async function (a
+        let partnerId;
+        if (id % 2 === 0) {
+            partnerId = 2;
+        } else {
+            partnerId = 3;
+        }
+        return {
+            author_id: [10, "Demo User"],
+            body: `<p>Message ${id + 1}</p>`,
+            date: "2019-04-20 10:00:00",
+            id: id + 1,
+            message_type: 'comment',
+            model: 'res.partner',
+            record_name: 'General',
+            res_id: partnerId,
+        };
+    });
+    this.data['mail.message'].records = messages;
+
+    await this.createView({
+        data: this.data,
+        hasView: true,
+        // View params
+        View: FormView,
+        model: 'res.partner',
+        arch: `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                    <field name="description"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>
+        `,
+        viewOptions: {
+            currentId: 2,
+            ids: [2, 3]
+        },
+        config: {
+            device: {size_class: 4}
+        }
+    });
+
+    const controllerContentEl = document.querySelector('.o_content');
+    const formViewEl = document.querySelector('.o_form_view');
+    assert.doesNotHaveClass(formViewEl, 'o_xxl_form_view',
+        "Form view is not xxl"
+    );
+
+    assert.strictEqual(
+        document.querySelector('.breadcrumb-item.active').textContent,
+        'second partner',
+        "'second partner' form view is opened"
+    );
+    assert.strictEqual(controllerContentEl.scrollTop, 0,
+        "The initial scrollTop of the form view is 0"
+    );
+    // scroll message list to a big value to get to the bottom
+    let scrollRecord1;
+    await afterNextRender(() => {
+        controllerContentEl.scrollTop = 9000;
+        scrollRecord1 = controllerContentEl.scrollTop;
+        return triggerEvent(
+            document.querySelector('.o_ThreadViewer_messageList'),
+            'scroll'
+        );
+    });
+    assert.ok(scrollRecord1 > 0, "We have scrolled the record to a non-zero value");
+    await afterNextRender(() =>
+        document.querySelector('.o_pager_next').click()
+    );
+    assert.strictEqual(
+        document.querySelector('.breadcrumb-item.active').textContent,
+        'third partner',
+        "'third partner' form view is opened"
+    );
+    assert.strictEqual(controllerContentEl.scrollTop, 0,
+        "The scrollTop of the form view is still 0 on another record"
+    );
+    await afterNextRender(() =>
+        document.querySelector('.o_pager_previous').click()
+    );
+    assert.strictEqual(controllerContentEl.scrollTop, 0,
+        "The scrollTop of the form view is still 0 on first record"
+    );
+});
+
+QUnit.test('Form view with primary button not scrolled when posting message', async function (assert) {
```

Comment:
what ? this is just how the form view is supposed to work (and one of the main aim of this PR -- and admittedly I just discovered it)
It has nothing to do with mail at all. In fact, when there is no primary button in the form's header, the scroll position stays around the message just posted.

Reply:
I'm not actually sure to understand what "primary button" is then, and why it should change the behavior.

1. For me the main issue to fix here is that when opening a view, any view, by default we should not scroll to a message at the bottom of it, but show the top of the form view. Most important elements are at the top, this is what we want to see by default for sure.

2. But when there is a specific user interaction with the chatter, such as posting a message, it is acceptable and even encouraged for me to actually scroll to the new message, because this is the only feedback the user is going to get.

Previously/currently, it was assumed that the only way to achieve 1. was to ignore 2. since 1. would clearly be considered more of a priority.
But if it is possible to do both 1. and 2. at the same time for me by all means it should be done, and I wouldn't like that we test against this in the meantime if there is a technical issue preventing it.

Reply:
@seb-odoo 
ha fuck, this is a regression from saas-13.3 at least, and already spotted by aju.

The reload of the form should only happen when followers have been added to the thread, and especially through the add followers feature of the composer.
In practice: CRM => create opportunity, create partner in the m2o with no email, fill in the email field of the crm.lead.
Click on send message
There should be a proposal to add the customer to the followers (and I think the side effect it to put email of the lead in the partner, not sure of that though)
Upon send, the form reloads

In other cases, it doesn't reload

Reply:
> In practice: CRM => create opportunity, create partner in the m2o with no email, fill in the email field of the crm.lead.
> Click on send message
> **There should be a proposal to add the customer to the followers** (and I think the side effect it to put email of the lead in the partner, not sure of that though)

task-2283356 FYI